contracts/helpers/TransferHelper.sol

```diff
@@ -324,36 +215,79 @@ contract TransferHelper is
         ) {
             // Check if the value returned from the external call matches
             // the conduit `execute` selector.
-            if (
-                conduitMagicValue != ConduitInterface(conduit).execute.selector
-            ) {
+            if (conduitMagicValue != ConduitInterface.execute.selector) {
                 // If the external call fails, revert with the conduit key
                 // and conduit address.
                 revert InvalidConduit(conduitKey, conduit);
             }
-        } catch (bytes memory data) {
-            // Catch reverts from the external call to the conduit and
-            // "bubble up" the conduit's revert reason.
-            revert ConduitErrorRevertBytes(data, conduitKey, conduit);
         } catch Error(string memory reason) {
             // Catch reverts with a provided reason string and
             // revert with the reason, conduit key and conduit address.
             revert ConduitErrorRevertString(reason, conduitKey, conduit);
+        } catch (bytes memory data) {
+            // Conduits will throw a custom error when attempting to transfer
+            // native token item types or an ERC721 item amount other than 1.
+            // Bubble up these custom errors when encountered. Note that the
+            // conduit itself will bubble up revert reasons from transfers as
+            // well, meaning that these errors are not necessarily indicative of
+            // an issue with the item type or amount in cases where the same
+            // custom error signature is encountered during a conduit transfer.
+
+            // Set initial value of first four bytes of revert data to the mask.
+            bytes4 customErrorSelector = bytes4(0xffffffff);
+
+            // Utilize assembly to read first four bytes (if present) directly.
+            assembly {
+                // Combine original mask with first four bytes of revert data.
+                customErrorSelector := and(
+                    mload(add(data, 0x20)), // Data begins after length offset.
+                    customErrorSelector
```

Comment:
Note: if `data` is an empty byte array, this could be reading junk value. Looked at the code generated: in the empty case, `data` will be `0x60`--the zero region in memory; So this would actually be reading the 0x80 or the very first free memory pointer location.

However, I don't really see this being problematic. In the worst case, the data at memory location `add(data, 0x20)` is the same as `InvalidItemType.selector` or `InvalidERC721TransferAmount.selector`, and then this would revert with that instead of the correct error `ConduitErrorRevertBytes`, i.e., the worst case is reverting with the wrong custom error.

Reply:
Yeah also data.length won’t be 4 in that case so the value read will just be ignored